src/deepsparse/loggers/function_logger.py

```diff
@@ -0,0 +1,137 @@
+
+         "pipeline_outputs":
+            [{"function": "identity",
+            "frequency": 4}]
+        }
+    """
+
+    def __init__(
+        self,
+        loggers: Union[BaseLogger, List[BaseLogger]],
+        config: Dict[str, List[Dict[str, Any]]],
+    ):
+
+        self.loggers = loggers if isinstance(loggers, list) else [loggers]
+        self.config = self._add_frequency_counter(config)
+
+    @property
+    def identifier(self) -> str:
+        return "function"
+
+    def log(self, identifier: str, value: Any, category: MetricCategories):
+        """
+        :param identifier: The identifier of the log
+             By default should consist of at least one string (name of the pipeline):
+                e.g. identifier = "pipeline_name"
+            If identifier is to be more complex, optional strings are to be concatenated
+            using dot as a separator:
+                e.g. identifier = "pipeline_name.some_argument_1.some_argument_2"
+
+            Note: by convention, "some_argument_1" very often is
+                  referred to as `target_name`
+        :param value: The data structure that the logger is logging
+        :param category: The metric category that the log belongs to
+        """
+        if category == MetricCategories.DATA:
+            pipeline_name, target_name, *_ = identifier.split(".")
+            list_functions = self.config.get(target_name)
+            if list_functions:
+                for (
+                    function_dict
+                ) in list_functions:  # iterate over all available functions
+                    if (
+                        function_dict["frequency_counter"] % function_dict["frequency"]
+                        == 0
+                    ):
+                        # if count matches the frequency, log
+                        # otherwise, pass
+                        self._log_data(identifier, value, category, function_dict)
+                    # increment the counter
+                    function_dict["frequency_counter"] += 1
+
+        else:
+            [
+                logger.log(identifier=identifier, value=value, category=category)
+                for logger in self.loggers
+            ]
+
+    def _log_data(
+        self,
+        identifier: str,
+        value: Any,
+        category: MetricCategories,
+        function_dict: Dict[str, Any],
+    ):
+        # fetch the string signature & apply function
+        # specified by the signature to the value
+        function_signature = function_dict.get("function")
+        value = apply_function(value=value, function_signature=function_signature)
+
+        # filter the loggers if required by the config
+        target_loggers = function_dict.get("target_logger")
+        if target_loggers:
+            valid_loggers = [
+                logger for logger in self.loggers if logger.identifier in target_loggers
+            ]
+        else:
+            valid_loggers = self.loggers
```

Comment:
ah hah, is this why you need list of loggers passed in?

this feels like something that should be handled elsewhere. but TBH let's double check that we actually want to support targeting a subset of loggers for certain values